### ISSUE TITLE
Incorrect dependencies in browser version of signalr-protocol-msgpack 1.0.0-preview1-final

### ISSUE THREAD

**Issue by AndriySvyryd** [aspnet/SignalR#1530]
Trying to bundle it using webpack produces:
>ERROR in ./node_modules/@aspnet/signalr-protocol-msgpack/dist/browser/signalr-protocol-msgpack.js
>Module not found : error : Can't resolve '    ode_modulesmsgpack5index.js' in 'D:\Programs\node_modules\\@aspnet\signalr-protocol-msgpack\dist\browser'
     @ ./node_modules/@aspnet/signalr-protocol-msgpack/dist/browser/signalr-protocol-msgpack.js 6:138-270
     @ dll vendor
ERROR in ./node_modules/@aspnet/signalr-protocol-msgpack/dist/browser/signalr-protocol-msgpack.js
Module not found : error : Can't resolve '../../node_modules/msgpack5/index.js' in 'D:\Programs\node_modules\\@aspnet\signalr-protocol-msgpack\dist\browser'
     @ ./node_modules/@aspnet/signalr-protocol-msgpack/dist/browser/signalr-protocol-msgpack.js 6:89-136
     @ dll vendor

webpack 3.10.0
node 8.9.3

**Comment by anurse** [comment]
You shouldn't really be using the `dist/browser` script from WebPack (though I would have expected it to work, so we should look at this). You should just be referencing `@aspnet/signalr-protocol-msgpack` as an ESM module. How are you importing the script in your entry-point?

**Comment by AndriySvyryd** [comment]
@anurse  I don't have any references to the browser version, I jut do `import { MessagePackHubProtocol } from '@aspnet/signalr-protocol-msgpack';`

I also tried using it as indendent without webpack via `<script src="/lib/signalr/signalr-protocol-msgpack.js"></script>` and it fails to load.

**Comment by WilStead** [comment]
I'm seeing this issue as well, and I'm also not making any direct references to the browser distribution anywhere in my own code, either.

In my webpack config, I'm using
```javascript
entry: {
    vendor: [
        '@aspnet/signalr',
        '@aspnet/signalr-protocol-msgpack'
    ]
}
```

In my script source, I have
```javascript
import { HubConnection, ProtocolType } from '@aspnet/signalr';
import { MessagePackHubProtocol } from '@aspnet/signalr-protocol-msgpack';
```

And in my tsconfig.json I have `"module": "commonjs"`

**Comment by AndriySvyryd** [comment]
Deleting the `node_modules\@aspnet\signalr-protocol-msgpack\dist\browser` folder allows webpack to work

**Comment by anurse** [comment]
Yeah, it looks like the UMD build of the msgpack library got borked up by our UMD process. The `rollup-plugin-commonjs` plug-in we use seems to mangle the paths and I'm not even sure it was working in the first place.

We don't have automation for the tests that ensure the UMD build of `@aspnet/signalr-protocol-msgpack` loads properly, and it looks like it got missed in verification (cc @muratg).

I'm surprised that WebPack is trying to load the `dist/browser` script. It's **not** labelled as the `main` script for the package, the CommonJS build (which should be working) is.

**Comment by anurse** [comment]
> Deleting the node_modules\@aspnet\signalr-protocol-msgpack\dist\browser folder allows webpack to work

Glad to hear you found a workaround. We're going to have to figure out how to resolve this (which means diving back into the chaos of JavaScript build tools...)

**Comment by anurse** [comment]
I think I have a fix. It turns out `rollup` is really sensitive to the **way** in which you flag external dependencies and changing from a function to a list of external modules actually changed to order of operations (gross).

**Comment by anurse** [comment]
This should be fixed in `1.0.0-preview1-update1` of `@aspnet/signalr-protocol-msgpack`. Please give it a try and let me know if you hit further issues. Filed #1542 because we don't intend for WebPack to use this version of the script but it is using it anyway :)

**Comment by AndriySvyryd** [comment]
@anurse Confirming new version is working as expected
